pkg/asset/machines/aws/machines.go

```diff
@@ -170,11 +170,30 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 	return machines, controlPlaneMachineSet, nil
 }
 
-func provider(in *machineProviderInput) (*machineapi.AWSMachineProviderConfig, error) {
+func provider(in *machineProviderInput, pool *aws.MachinePool) (*machineapi.AWSMachineProviderConfig, error) {
 	tags, err := tagsFromUserTags(in.clusterID, in.userTags)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create machineapi.TagSpecifications from UserTags")
 	}
+
+	securityGroups := []machineapi.AWSResourceReference{{
+		Filters: []machineapi.Filter{{
+			Name:   "tag:Name",
+			Values: []string{fmt.Sprintf("%s-%s-sg", in.clusterID, in.role)},
+		}},
+	}}
+	securityGroupsFromPool := []machineapi.AWSResourceReference{}
+	for _, sgID := range pool.AdditionalSecurityGroupIDs {
+		sgID := sgID
```

Comment:
Do we need this temporary variable sgID (the one on the left of := ) ?

Reply:
@sadasu yes because we're storing the address of the `sgID` in the following lines and the outer `sgID` pointed to value changes at each iteration of the loop. Without this, the linter will complain as well.